daemon/metrics.go

```diff
@@ -0,0 +1,33 @@
+package daemon
+
+import "github.com/docker/go-metrics"
+
+var (
+	containerActions   metrics.LabeledTimer
+	imageActions       metrics.LabeledTimer
+	daemonActions      metrics.LabeledTimer
+	networkActions     metrics.LabeledTimer
+	engineVersion      metrics.LabeledGauge
+	engineCpus         metrics.Gauge
+	engineMemory       metrics.Gauge
+	healthCheckCounter metrics.LabeledCounter
+)
+
+func init() {
+	ns := metrics.NewNamespace("engine", "daemon", nil)
+	containerActions = ns.NewLabeledTimer("container_actions", "The number of seconds it takes to process each container action", "action")
```

Comment:
These should be initialised with all possible values for 'action' to avoid weirdness. This is done usually by `containerActions.WithLabelValues("xxx")`